library/std/src/sys/hermit/mod.rs

```diff
@@ -76,9 +76,24 @@ pub fn abort_internal() -> ! {
     }
 }
 
-// FIXME: just a workaround to test the system
 pub fn hashmap_random_keys() -> (u64, u64) {
-    (1, 2)
+    let mut buf = [0; 16];
+    let mut slice = &mut buf[..];
+    while !slice.is_empty() {
+        let res = unsafe { abi::read_entropy(slice.as_mut_ptr(), slice.len(), 0) };
+        if res < 0 {
+            panic!(
+                "random key generation failed: {}",
+                crate::io::Error::from_raw_os_error(-res as i32)
```

Comment:
I'm not familiar with hermit error codes. Could you briefly explain why `-res as i32`?